sdk/core/azure-core/src/main/java/com/azure/core/util/UrlBuilder.java

```diff
@@ -159,13 +179,58 @@ public UrlBuilder setQuery(String query) {
         return this;
     }
 
+    /**
+     * Clear the query that will be used to build the final URL.
+     *
+     * @return This UrlBuilder so that multiple setters can be chained together.
+     */
+    public UrlBuilder clearQuery() {
+        if (query != null && !query.isEmpty()) {
+            query.clear();
+        }
+        return this;
+    }
+
     /**
      * Get the query that has been assigned to this UrlBuilder.
      *
      * @return the query that has been assigned to this UrlBuilder.
      */
     public Map<String, String> getQuery() {
-        return query;
+        // This contains a map of key=value query parameters, replacing
+        // multiple values for a single key with the first value in the list
+        // This is done to maintain backward compatibility.
+        final Map<String, String> singleKeyValueQuery =
+            this.query.entrySet()
+                      .stream()
+                      .collect(Collectors.toMap(
+                            e -> e.getKey(),
+                            e -> {
+                                QueryParameter parameter = e.getValue();
+                                String value = null;
+
+                                if (parameter != null) {
+                                    // return only the first in the list of multiple
+                                    // values for the same named parameter.
+                                    // e.g. name=a&name=b, returns name=a
+                                    value = parameter.getValuesList().get(0);
```

Comment:
Even though it is a functional change, I think this should return the full query parameter. Have this only return the first value may result in errors, or confusing runtime scenarios.

Reply:
Changed to return "name=value1,value2,value3" as suggested in the comments above.